src/server/getCurrentUser.ts

```diff
@@ -0,0 +1,17 @@
+"use server";
+
+import { getSupabaseServerComponentClient } from "@/supabaseServer";
+import { logErrorReturnLogId } from "@/logger/logger";
+import { User } from "@supabase/gotrue-js";
+
+export async function getCurrentUser(): Promise<User | null> {
+    const serverComponentClient = getSupabaseServerComponentClient();
+
+    const { data, error } = await serverComponentClient.auth.getUser();
```

Comment:
nitpick: just for consistency
```suggestion
    const supabase = getSupabaseServerComponentClient();

    const { data, error } = await supabase.auth.getUser();
```